docs/DeveloperGuide.md

```diff
@@ -1,38 +1,353 @@
+  ![img.png](CommandHandler.png)
+- `FileHandler`: Reads data from, and writes data to, the hard disk.
+  ![img.png](FileHandler.png)
+- `FindVisit`: The command executor handling visits related work.
+  ![img.png](FindVisit.png)
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The entities storing data are the following components:
+- `Records`: The list of patients.
+  ![img.png](Records.png)
+- `AppointmentRecord`: The list of appointments.
+  ![img.png](AppointmentRecord.png)
+- `Appointment` : Information of appointment.
+  ![img.png](Appointment.png)
+- `Visit`: The visit of patients.
+  ![img.png](Visit.png)
+- `Patient`: Personal information of patient.
+  ![img.png](Patient.png)
 
-## Non-Functional Requirements
 
-{Give non-functional requirements}
+### 1. Managing Patient Visits and Records
+### a. Adding New Visits for Existing Patients
+The addVisit mechanism is handled by `CommandHandler`. It begins by scanning the patient records (`Records`) to locate 
+the patient with the specified NRIC. Once the target patient is found, a new `Visit` object is created using the 
+provided visit details. This new visit is then added to the patient's `ArrayList<Visit>`. Finally, the updated patient 
+records are saved using `FileHandler` to ensure they can be retrieved later.
+
+**Doctor enters the command:** `addVisit ic/S9870789B v/29-10-2024 10:00 d/Gastric m/Gaviscon, Paracetamol`
+
+> The Object Diagram before the execution of addVisit command:
+![img.png](ObjectDiagramBeforeAddVisit.png)
+
+ > The Object Diagram after the execution of addVisit command:
+![img.png](ObjectDiagramAfterAddVisit.png)
+
+> The Sequence Diagram for the execution of addVisit command:
+![img.png](AddVisitSequenceDiagram.png)
+
+### b. Adding New Patient to the Patient Records
+The "add" command mechanism for New Patients(first visit to clinic) is handled by `CommandHandler`. It begins by creating a new `Patient` object to store all related
+information from terminal input. A new `Visit` object is compulsory to be created with the `Patient` object. This new 
+visit is then added to the patient's `ArrayList<Visit>`. Finally, the new patient will be stored into the patient 
+records (`Records`) which will be saved using `FileHandler` to ensure they can be retrieved later.
+
+**Doctor enters the command:** `add n/Patricia Chan ic/S9870789B p/98097890 d/Cough m/Antibiotics ha/Bukit Batok East Avenue 3 
+dob/01111998 v/21-10-2024 18:00 al/Peanuts s/Female mh/Hypertension`
+
+> The Object Diagram before the execution of "add" command:
+![img.png](ObjectDiagramBeforeAddPatient.png)
+
+> The Object Diagram after the execution of "add" command:
+![img.png](ObjectDiagramAfterAddPatient.png)
```

Comment:
It is not blatantly clear what is being changed, too much detail. Maybe it would be beneficial to put the diagrams side by side?